File: Bachelor/Semester4/Systems_for_Design_and_Implementation/Java/src/test/java/domain/validator/TripValidatorTest.java
package domain.validator;

import domain.Trip;
import junit.framework.TestCase;

import java.time.LocalDateTime;

public class TripValidatorTest extends TestCase {

    public void testValidate() {
        TripValidator validator = new TripValidator();
        Trip trip = new Trip(1L, "Paris","Auto", LocalDateTime.MIN, 129, 20);
        validator.validate(trip);
        trip = new Trip(2L,"","",LocalDateTime.MAX,-42,-1);
        try{
            validator.validate(trip);
            fail();
        }catch (ValidationException e){
            assertEquals("Invalid destination\nInvalid transport firm\nInvalid price\nInvalid number of seats\n",e.getMessage());
        }
    }
}

File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Tests/domain/validator/TripValidatorTest.cs
using System;
using NUnit.Framework;
using ProiectCShMPP.domain;
using ProiectCShMPP.domain.validator;

namespace Tests.domain.validator
{
    [TestFixture]
    public class TripValidatorTest
    {
        private readonly IValidator<long, Trip> _validator = new TripValidator();
        
        [Test]
        public void TestValidate()
        {
            Trip trip = new Trip(1L)
            {
                Destination = "Cairo",
                DepartureTime = new DateTime(2021, 1, 1, 1, 1, 1),
                Price = 120,
                Seats = 10,
                TransportFirm = "A .SRL"
            };
            _validator.Validate(trip);
            try
            {
                _validator.Validate(new Trip(2L)
                {
                    Destination = "",
                    DepartureTime = DateTime.MaxValue,
                    Price = -32,
                    Seats = 10,
                    TransportFirm = "IO"
                });
                Assert.Fail();
            }
            catch (ValidationException e)
            {
                Assert.AreEqual("Invalid destination\nInvalid price\n", e.Message);
            }
        }
    }
}